### PR TITLE
Add checking for finalized consistency

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds the missing p2p check:
```
[REJECT] The current finalized_checkpoint is an ancestor of the block defined by aggregate.data.beacon_block_root -- i.e. get_ancestor(store, aggregate.data.beacon_block_root, compute_start_slot_at_epoch(store.finalized_checkpoint.epoch)) == store.finalized_checkpoint.root
```

The check was introduced in v0.12.3 but we somehow missed it

### CODE REVIEW

File: beacon-chain/blockchain/receive_attestation.go [hunk 1]
```diff
@@ -96,6 +97,11 @@ func (s *Service) VerifyLmdFfgConsistency(ctx context.Context, a *ethpb.Attestat
 	return s.verifyLMDFFGConsistent(ctx, a.Data.Target.Epoch, a.Data.Target.Root, a.Data.BeaconBlockRoot)
 }
 
+// VerifyFinalizedConsistency verifies input root is consistent with finalized store.
+func (s *Service) VerifyFinalizedConsistency(ctx context.Context, r []byte) error {
+	return s.verifyFinalizedConsistency(ctx, r)
+}
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/receive_attestation.go:103]: Why have a public method that is just passthrough to the private method?

File: beacon-chain/blockchain/process_attestation_helpers.go [hunk 2]
```diff
@@ -178,6 +178,24 @@ func (s *Service) verifyLMDFFGConsistent(ctx context.Context, ffgEpoch uint64, f
 	return nil
 }
 
+// verifyFinalizedConsistency verifies input root is consistent with finalized store.
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/process_attestation_helpers.go:181]: Can you add more commentary to this. Why might this validation fail?

File: beacon-chain/blockchain/process_attestation_test.go [hunk 3]
```diff
@@ -491,6 +491,38 @@ func TestVerifyLMDFFGConsistent_OK(t *testing.T) {
 	assert.NoError(t, err, "Could not verify LMD and FFG votes to be consistent")
 }
 
+func TestVerifyFinalizedConsistency_OK(t *testing.T) {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/process_attestation_test.go:494]: Please also test the unhappy path.

Comment (terencechain) [reply beacon-chain/blockchain/process_attestation_test.go:494]: Unhappy path was tested in this func but I'll move it to a new test for clarity

See line 518

File: beacon-chain/blockchain/receive_attestation.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/blockchain/receive_attestation.go:103]: I was following `VerifyLmdFfgConsistency`'s pattern but it was a bad follow. I'll fix it

**Review by rauljordan — APPROVED** [review]
lgtm, let's check at runtime

**Pull request merged**